src/test/java/seedu/address/model/InternshipDiaryTest.java

```diff
@@ -0,0 +1,106 @@
+package seedu.address.model;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static seedu.address.testutil.Assert.assertThrows;
+import static seedu.address.testutil.TypicalInternshipApplications.GOOGLE;
+import static seedu.address.testutil.TypicalInternshipApplications.getTypicalInternshipDiary;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+
+import org.junit.jupiter.api.Test;
+
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import seedu.address.model.internship.InternshipApplication;
+import seedu.address.model.internship.exceptions.DuplicateInternshipApplicationException;
+import seedu.address.testutil.InternshipApplicationBuilder;
+
+
+
+public class InternshipDiaryTest {
+
+    private final InternshipDiary internshipDiary = new InternshipDiary();
+
+    @Test
+    public void constructor() {
+        assertEquals(Collections.emptyList(), internshipDiary.getInternshipList());
+    }
+
+    @Test
+    public void resetData_null_throwsNullPointerException() {
+        assertThrows(NullPointerException.class, () -> internshipDiary.resetData(null));
+    }
+
+    @Test
+    public void resetData_withValidReadOnlyInternshipDiary_replacesData() {
+        InternshipDiary newData = getTypicalInternshipDiary();
+        internshipDiary.resetData(newData);
+        assertEquals(newData, internshipDiary);
+    }
+
+    @Test
+    public void resetData_withDuplicateInternship_throwsDuplicateInternshipException() {
```

Comment:
I think this checks to see if resetData properly throws exception when there are two similar internship application